src/lib/network_pool/indexed_pool.ml

```diff
@@ -1104,7 +1104,9 @@ let%test_module _ =
                 [%test_eq:
                   Transaction_hash.User_command_with_valid_signature.t
                   Sequence.t] dropped' (Sequence.singleton cmd) ;
-                [%test_eq: t] pool pool''
+                (* TODO: derive compare, equal for t *)
+                let compare = Caml.compare in
```

Comment:
Does this `compare` get used by the expanded macro on the next line or something?

Reply:
This was an annoying one: the macro assumed that `compare : t -> t -> int` existed, but we weren't deriving `compare` for `t` because `F_sequence.t` is a GADT.

I added a manual compare implementation for `F_sequence.t` and added the `deriving compare` above, so we no longer need this.